src/services/DeviceService.ts

```diff
@@ -0,0 +1,60 @@
+import winston from 'winston';
+import DeviceModel from '../models/device';
+
+export default class DeviceService {
+  // region public static methods
+  public static async addDevice(mail: string, token: string) {
+    let device = await DeviceModel.findOne({ mail });
+    if (!device) {
+      device = await DeviceService.save(mail);
```

Comment:
I would prefer not to use the additional `.save` function and construct the object here. This would improve readability.